### PR TITLE
Add IAM Role support for jclouds integration.

### PR DESCRIPTION
**Pull request by kobalski**
This pr makes call AWS API defined here http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials and parses and builds `Credentials` for JClouds API. Parser code is copied from `hazelcast-clouds` module.

### CODE REVIEW

File: hazelcast-jclouds/src/main/java/com/hazelcast/jclouds/ComputeServiceBuilder.java [hunk 1]
```diff
@@ -144,7 +146,6 @@ ComputeService build() {
         final String identity = getOrNull(JCloudsProperties.IDENTITY);
         String credential = getOrNull(JCloudsProperties.CREDENTIAL);
         final String credentialPath = getOrNull(JCloudsProperties.CREDENTIAL_PATH);
-
         checkNotNull(cloudProvider, "Cloud provider is not set");
         checkNotNull(identity, "Cloud provider identity is not set");
```
Comment (bilalyasar) [inline hazelcast-jclouds/src/main/java/com/hazelcast/jclouds/ComputeServiceBuilder.java]: what if user only set `iam-role` ... in that case , user doesn't have to set `identity`. right?

Comment (kobalski) [inline hazelcast-jclouds/src/main/java/com/hazelcast/jclouds/ComputeServiceBuilder.java]: Yes, right moved checks here : https://github.com/hazelcast/hazelcast/pull/6476/files#diff-627080c6781ee2f36b9958aafc799128R256 If iam-role is set no identity or credential should be set. If aim-role is not set identity or credentials must be present.

**Comment by kobalski** [comment]
verify

**Comment by bilalyasar** [comment]
:+1: , once 💚

**Comment by kobalski** [comment]
verify

**Pull request merged**